x-pack/solutions/search/plugins/search_playground/public/utils/stream.ts

```diff
@@ -5,182 +5,58 @@
  * 2.0.
  */
 
-import type { JSONValue } from '../types';
+import type { UIMessageChunk } from 'ai';
 
-export interface StreamPart<CODE extends string, NAME extends string, TYPE> {
-  code: CODE;
-  name: NAME;
-  parse: (value: JSONValue) => { type: NAME; value: TYPE };
+interface ReadDataStreamOptions {
+  isAborted?: () => boolean;
 }
 
-type StreamParts =
-  | typeof textStreamPart
-  | typeof errorStreamPart
-  | typeof messageAnnotationsStreamPart;
-/**
- * Maps the type of a stream part to its value type.
- */
-type StreamPartValueType = {
-  [P in StreamParts as P['name']]: ReturnType<P['parse']>['value'];
-};
-
-export type StreamPartType =
-  | ReturnType<typeof textStreamPart.parse>
-  | ReturnType<typeof errorStreamPart.parse>
-  | ReturnType<typeof messageAnnotationsStreamPart.parse>
-  | ReturnType<typeof bufferStreamPart.parse>;
-
-const NEWLINE = '\n'.charCodeAt(0);
-
-const concatChunks = (chunks: Uint8Array[], totalLength: number) => {
-  const concatenatedChunks = new Uint8Array(totalLength);
-
-  let offset = 0;
-  for (const chunk of chunks) {
-    concatenatedChunks.set(chunk, offset);
-    offset += chunk.length;
-  }
-  chunks.length = 0;
-
-  return concatenatedChunks;
-};
-
 export async function* readDataStream(
   reader: ReadableStreamDefaultReader<Uint8Array>,
-  { isAborted }: { isAborted?: () => boolean } = {}
-): AsyncGenerator<StreamPartType> {
+  { isAborted }: ReadDataStreamOptions = {}
+): AsyncGenerator<UIMessageChunk> {
   const decoder = new TextDecoder();
-  const chunks: Uint8Array[] = [];
-  let totalLength = 0;
+  let buffer = '';
 
   while (true) {
-    const { value } = await reader.read();
+    const { done, value } = await reader.read();
 
     if (value) {
-      chunks.push(value);
-      totalLength += value.length;
-      if (value[value.length - 1] !== NEWLINE) {
-        continue;
-      }
-    }
+      buffer += decoder.decode(value, { stream: true });
 
-    if (chunks.length === 0) {
-      break;
-    }
+      let separatorIndex = buffer.indexOf('\n\n');
```

Comment:
We were using a constant `NEWLINE` for this previously, is the change to 2 newlines because of the dependency update?

Should we still have a constant for this?